lib/assets/javascripts/turbolinks.coffee

```diff
@@ -200,9 +210,26 @@ swapNodes = (targetBody, existingNodes, options) ->
         existingNode = targetNode.ownerDocument.adoptNode(existingNode)
         targetNode.parentNode.replaceChild(existingNode, targetNode)
       else
-        existingNode.parentNode.replaceChild(targetNode, existingNode)
-        onNodeRemoved(existingNode)
-        changedNodes.push(targetNode)
+        if options.append or options.prepend
+          ownerDocument = existingNode.ownerDocument
+          firstChild = existingNode.firstChild
+
+          childNodes = Array::slice.call targetNode.childNodes, 0 # a copy has to be made since the list is mutated while processing
+
+          for childNode in childNodes
+            ownerDocument.adoptNode(childNode)
```

Comment:
`adoptNode` has bad side effects in IE (textarea placeholder bug - https://github.com/rails/turbolinks/commit/d67a2d5ad8680ce4a653242840d9300206c2a997) and Chrome/Safari (double-eval scripts - https://github.com/rails/turbolinks/commit/27318d2f0ab37e0abfee405e1bb85dde7d90f226). On L210 I believe it's required for DOM data (event listeners &co) to be transferred properly, but here we can skip it because we only care about the markup. 

The spec says we should call `importNode` / `adoptNode` before inserting a node into a different document, but in reality no browsers actually enforce this (see http://devdocs.io/dom/document/adoptnode#Notes).